src/mrinufft/operators/interfaces/finufft.py

```diff
@@ -141,12 +149,13 @@ def __init__(
             n_batchs=n_batchs,
             n_trans=n_trans,
             smaps=smaps,
+            raw_op=self.raw_op,
             squeeze_dims=squeeze_dims,
         )
 
-        self.raw_op = RawFinufftPlan(
-            samples,
-            shape,
-            n_trans=n_trans,
-            **kwargs,
-        )
+    @FourierOperatorBase.samples.setter
+    def samples(self, samples):
+        """Update the plans when changing the samples."""
+        self._samples = samples
+        for typ in [1, 2, "grad"]:
```

Comment:
Now I see why you initialized _make_grad_plan above. However, we should update points for grad_plans only if grad_wrt_traj is set to True.